### ISSUE TITLE
Board Mockup Arduino

### ISSUE THREAD

**Issue by mullets** [tomaskovacik/VAGFISWriter#10]
First congratulations for the project! I would like to try to do the same with my panel, could you post on how to connect a Mockup of the Arduino board with the resistors? I'm really confused about the call and I believe it would help a lot of people! Thank you and congratulations again.

**Comment by tomaskovacik** [comment]
Simply connect enable, data an clock pins from cluster to defined pin in example (FIS_ENA , FIS_CLK,FIS_DATA) and you are good to go. Do not forget ground connection.

For no 5V mcu:
https://github.com/tomaskovacik/VAGFISWriter/wiki/How-to-connect

**Comment by mullets** [comment]
Yes! But how connect the Gnd? (I see the picture but I don't know is mcu)

**Comment by tomaskovacik** [comment]
connect grounds together.

https://en.wikipedia.org/wiki/Microcontroller

**Comment by mullets** [comment]
It's ok! so I connect gnd from Arduino to a 4k7 resistor and I connect it to the panel ENA?

Sorry my English, I'm from Brazil.

**Comment by mullets** [comment]
![example](https://user-images.githubusercontent.com/1914690/105926796-72bb5900-6021-11eb-96fc-88f8d26132d9.png)

I have tried this, but it doesn't work. My panel is on the table, outside the car, to test the system before assembling. If you can help me, thank you.

**Comment by matt2005** [comment]
Does the panel power up?

**Comment by mullets** [comment]
> Does the panel power up?

Yes! The panel is perfect works and power, without erros in FIS.. 
![4f2e3fa3-e9bf-4328-b556-522cb11ffbc8](https://user-images.githubusercontent.com/1914690/105996035-94036000-6088-11eb-9e61-990d7d587d7e.jpg)

**Comment by tomaskovacik** [comment]
Remove resistors, they are not needed:

![20210127_141534.jpg](https://user-images.githubusercontent.com/4346464/105996201-4e578f00-60aa-11eb-9332-da231a96bde8.jpg)

Usin ping 567 on nano

T.

**Comment by mullets** [comment]
This is my pinout
BLUE 1998, 1999
1 - Terminal 15
2 - Right turn signal
3 - Speedometer output 1
4 - Vacant
5 - Fuel gauge sender
6 - Airbag
7 - Terminal 31 sensor Ground (GND)
8 - Coolant temperature
9 - Terminal 31 load Ground (GND)
10 - Oil pressure switch
11 - Engine speed (RPM) signal
12 - Terminal 61
13 - CAT
14 - Self-levelling suspension
15 - Terminal 58d
16 - Malfunction Indicator Lamp (MIL)
17 - High beams
18 - Left turn signal
19 - ABS
20 - Terminal 58s
21 - Door contact, driver's side
22 - Coolant level low
23 - Terminal 30
24 - Terminal 31 load Ground (GND)
25 - K-wire
26 - Right parking light
27 - Left parking light
28 - Speedometer input
29 - Brake fluid level/pressure
30 - S-contact
31 - Seat belt buckle
32 - Electronic Stability Program (ESP) 


Green 1998, 1999
1 - Vacant
2 - Transponder 1
3 - Vacant
4 - Immobilizer
5 - W-wire
6 - Vacant
7 - Brake pads
8 - Input for outside buzzer control (currently vacant)
9 - Input for outside gong/chime control (currently vacant)
10 - Fuel tank warning output signal for Engine Control Module (ECM)
11 - Standing time output
12 - Air conditioning shut-off
13 - Parking brake
14 - Electronic throttle
15 - Side marker light
16 - Vacant
17 - Transponder 2
18 - Vacant
19 - Vacant
20 - Vacant
21 - Oil temperature
22 - Vacant
23 - Vacant
24 - Vacant
25 - Vacant
26 - Vacant
27 - Vacant
28 - Vacant
29 - Terminal 58d (outside dimmer potentiometer)
30 - Speedometer output 2
31 - Vacant
32 - Vacant


RED 1998, 1999
1 - Fuel consumption signal
2 - Vacant
3 - Vacant
4 - Vacant
5 - Outside air temperature
6 - Selector lever position indicator
7 - Vacant
8 - Vacant
9 - Washer fluid
10 - Hydraulic pressure
11 - On-board computer reset
12 - Clock signal for radio frequency display
13 - Data signal for radio frequency display
14 - On-board computer, left
15 - Enable-signal for radio frequency display
16 - Tail lights / dimmed headlights (bulb monitor)
17 - Vacant
18 - On-board computer, right
19 - Brake lights
20 - Vacant

> Remove resistors, they are not needed:
> 
> ![20210127_141534.jpg](https://user-images.githubusercontent.com/4346464/105996201-4e578f00-60aa-11eb-9332-da231a96bde8.jpg)
> 
> Usin ping 567 on nano
> 
> T.

Ok! Without resistors, dont work :/ 

The solutions is buy one arduino nano?  I have a lot of anothers Uno boards here, really need buy one nano? 

Thanks T.

**Comment by mullets** [comment]
In your image you have 4 cables connected, E D C and another one, where does the other one go on the panel?

> Remove resistors, they are not needed:
> 
> ![20210127_141534.jpg](https://user-images.githubusercontent.com/4346464/105996201-4e578f00-60aa-11eb-9332-da231a96bde8.jpg)
> 
> Usin ping 567 on nano
> 
> T.

**Comment by mullets** [comment]
> Does the panel power up?

my panel 
![ec7d2072-1335-4e28-9f45-677d6bbbbd80](https://user-images.githubusercontent.com/1914690/105996903-b6e24400-6089-11eb-916f-ae7847f034de.jpg)

**Comment by tomaskovacik** [comment]
no no, it must work. ok then few questions:
 do you have cluster coded for using navigation? or your stock radio is displaying text on upper part of screen?

on cluster you are using, correct?
RED 1998, 1999
12 - Clock signal for radio frequency display
13 - Data signal for radio frequency display
15 - Enable-signal for radio frequency display
?

it is possible that you have cluster in radio mode: https://github.com/tomaskovacik/VAGFISWriter/wiki/Theory-of-operation

to test it try this skech:

```
#include "VAGFISWriter.h"
#define FIS_ENA 5
#define FIS_CLK 6
#define FIS_DATA 7

VAGFISWriter fisWriter( FIS_CLK, FIS_DATA, FIS_ENA );

void setup() {
  fisWriter.begin();
  fisWriter.reset();
}
long time = 0;
void loop() {

  fisWriter.sendMsg("TEST12345678TEST);
  delay(1000);
}
```

**Comment by tomaskovacik** [comment]
> In your image you have 4 cables connected, E D C and another one, where does the other one go on the panel?

you mean ground? :)

**Comment by mullets** [comment]
> do you have cluster coded for using navigation? or your stock radio is displaying text on upper part of screen?
**My panel is stock radio display text upper part of screen! (the version with NAV didn't come in Brazil)**

> on cluster you are using, correct?
RED 1998, 1999
12 - Clock signal for radio frequency display
13 - Data signal for radio frequency display
15 - Enable-signal for radio frequency display
?

**Yes!**

>to test it try this skech:
**Don't work :(*

**Comment by tomaskovacik** [comment]
> My panel is stock radio display text upper part of screen! (the version with NAV didn't come in Brazil)

so then that's the problem :) go on and add 16 to coding of your cluster  but that sketch should work (maybe I should check radio mode part of code, it may be broken)

here is how to code your cluster:

http://wiki.ross-tech.com/wiki/index.php/Audi_A4_(8D)_Instrument_Cluster

**Comment by mullets** [comment]
> > My panel is stock radio display text upper part of screen! (the version with NAV didn't come in Brazil)
> 
> so then that's the problem :) go on and add 16 to coding of your cluster but that sketch should work (maybe I should check radio mode part of code, it may be broken)
> 
> here is how to code your cluster:
> 
> http://wiki.ross-tech.com/wiki/index.php/Audi_A4_(8D)_Instrument_Cluster


Thanks!

Do you have any suggestions on which setting to put?

**Comment by tomaskovacik** [comment]
> Do you have any suggestions on which setting to put?

take what ever number you have there and add 16000 to it

**Comment by mullets** [comment]
Ok! im try and talk with you in a couple of hours, thanks man!!!

**Comment by mullets** [comment]
Do you know if I can connect the vcds directly to the dashboard without having to turn it on in the car?

**Comment by tomaskovacik** [comment]
> 
> Do you know if I can connect the vcds directly to the dashboard without having to turn it on in the car?

Of corse, simply search for pinout of ond2 plug, tak cable apart and connect 12V ground and kline signal.  kline goes to cluster directly from adapter to blue connector pin 25

**Comment by mullets** [comment]
![image](https://user-images.githubusercontent.com/1914690/106001517-d16aec00-608e-11eb-943f-2ec230bb91a1.jpeg)

**Comment by tomaskovacik** [comment]
I used the wrong function in test code for radio mode in the previous post, if anyone search this issue it is alwary fixed

function should be sendRadioMsg()

**Comment by mullets** [comment]
> sendRadioMsg

Now WORKS!

**Comment by tomaskovacik** [comment]
> > sendRadioMsg
> 
> Now WORKS!

ok, and full screen mode?

**Comment by mullets** [comment]
First I just changed the code line and it worked on the first 2 lines (instead of the radio)

Then I made the change in VCDS to 16, and now it's perfect!

Thank you very much T.! And congratulations on the work.

Now I'm going after a FTK32RL KKL cable! (I already bought 3 different ones on Ali and none of them had the chip)

Thanks again.
![011e9442-00c3-4bfe-be9f-65aa58232360](https://user-images.githubusercontent.com/1914690/106037580-a47dff80-60b5-11eb-998d-35d97593d308.jpg)
